apps/frontend/src/app/5_pages/MarketMakingPage/components/AdjustAndDepositModal/components/NewPoolStatistics/NewPoolStatistics.tsx

```diff
@@ -37,16 +44,74 @@ export const NewPoolStatistics: FC<NewPoolStatisticsProps> = ({
   adjustType,
   pool,
 }) => {
-  const token = useMemo(() => pool.assetA, [pool.assetA]);
+  const tokenA = useMemo(() => pool.assetA, [pool.assetA]);
+  const { balanceA, balanceB } = useGetUserInfo(pool);
+  const { price: rbtcPrice } = useGetRBTCPrice();
+  const { reward } = useGetAccumulatedReward(pool.poolTokenA);
+  const { reward: accumulatedRewards } = useGetUserInfo(pool);
 
-  const newPoolBalance = useMemo(
+  const renderRewards = useMemo(
+    () =>
+      fromWei(decimalic(reward).add(decimalic(accumulatedRewards)).toString()),
+    [reward, accumulatedRewards],
+  );
+  const [weeklyRewardsEstimation, setWeeklyRewardsEstimation] = useState('0');
+
+  const { amount: expectedTokenAmount } = useGetExpectedTokenAmount(
+    pool,
+    amount,
+  );
+
+  const newPoolBalanceA = useMemo(
     () =>
       adjustType === AdjustType.Deposit || isInitialDeposit
-        ? initialPoolBalance.add(amount)
-        : initialPoolBalance.sub(amount),
-    [adjustType, amount, isInitialDeposit],
+        ? decimalic(balanceA).add(decimalic(amount))
```

Comment:
You have so many unnecessary castings in this component, you use `decimalic` almost everywhere. Once you fix the `useGetUserInfo` hook and its typings, make sure you cast only if necessary, it shouldn't be the default action. Bear in mind we want to use `Decimal` for everything and cast it to `Number`/`String` only if components need it for displaying something.

If you use it incorrectly, it clutters components/utils/hooks, makes them harder to analyze and harder to debug in case there's some bug.